src/Compilers/Core/Portable/CommandLine/CommonCompiler.cs

```diff
@@ -376,7 +372,30 @@ private int RunCore(TextWriter consoleOutput, ErrorLogger errorLogger, Cancellat
 
             try
             {
-                Func<DiagnosticBag, bool, bool> getAnalyzerDiagnostics = null;
+                // Print the diagnostics produced during the parsing stage and exit if there were any errors.
+                if (ReportErrors(compilation.GetParseDiagnostics(), consoleOutput, errorLogger))
+                {
+                    return Failed;
+                }
+
+                if (!sourceGenerators.IsEmpty)
+                {
+                    // TODO: Generated source path should include a "GeneratedFiles" subdirectory
+                    // so that "build clean" can be modified to delete the entire directory.
+                    var trees = compilation.GenerateSource(sourceGenerators, this.Arguments.OutputDirectory, writeToDisk: true, cancellationToken: cancellationToken);
+                    if (!trees.IsEmpty)
+                    {
+                        compilation = compilation.AddSyntaxTrees(trees);
+                        // Are there parse warnings? If so, we're probably reporting warnings
+                        // from the previous GetParseDiagnostics() call again here.
```

Comment:
Does this need a follow-up? Do we have an issue tracking this work? I suppose it could be a TODO comment without an issue instead.